jmolecules-architecture/jmolecules-cqrs-architecture/src/main/java/org/jmolecules/architecture/cqrs/annotation/ReadModel.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jmolecules.architecture.cqrs.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Identifies a read model element in the context of CQRS, i.e. a (persistent) object optimized for read-access and only
+ * only on the Q(uery) part of the architecture. The read model represents the current state or rather materialized view
+ * after replaying the events for a single entity.
+ *
+ * @author Christian Stettler
+ * @author Henning Schwentner
+ * @author Stephan Pirnbaum
+ * @author Martin Schimak
+ * @author Oliver Drotbohm
+ * @see <a href="http://cqrs.files.wordpress.com/2010/11/cqrs_documents.pdf">CQRS Documents by Greg Young</a>
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Documented
+public @interface ReadModel {
```

Comment:
As described in [my comment on the issue](https://github.com/xmolecules/jmolecules/issues/18#issuecomment-747929838), I'd like to make a case to rename this to `QueryModel`. It sounds more in line with *CQ*RS, rather than implying Read-Write Responsibility Segregation.

Reply:
We seem to have had settled on `ReadModel` in #18 but I see the argument. I am not opposed to rather go with `QueryModel` to align with the Q in CQRS.

Reply:
Then it'll be query model.